saleor/order/actions.py

```diff
@@ -428,126 +454,100 @@ def _get_fulfillment_line_if_exists(
-        refunded_line.quantity += unfulfilled_to_refund
+        # calculate the quantity fulfilled/unfulfilled to move
+        unfulfilled_to_move = min(line_to_move.quantity_unfulfilled, quantity_to_move)
+        quantity_to_move -= unfulfilled_to_move
+        line_to_move.quantity_fulfilled += unfulfilled_to_move
+        moved_line.quantity += unfulfilled_to_move
 
         # update current lines with new value of quantity
-        order_lines_to_update.append(line_to_refund)
+        order_lines_to_update.append(line_to_move)
 
-        refund_amount += line_to_refund.unit_price_gross_amount * unfulfilled_to_refund
-        if refunded_line.quantity > 0 and not fulfillment_line_existed:
-            # If this is new type of (orderl_line, stock) then we create new fulfillment
+        if moved_line.quantity > 0 and not fulfillment_line_existed:
+            # If this is new type of (order_line, stock) then we create new fulfillment
             # line
-            fulfillment_lines_to_create.append(refunded_line)
+            fulfillment_lines_to_create.append(moved_line)
         elif fulfillment_line_existed:
-            # if refund fulfillment already have the same line, we just update the
+            # if target fulfillment already have the same line, we just update the
             # quantity
-            fulfillment_lines_to_update.append(refunded_line)
+            fulfillment_lines_to_update.append(moved_line)
 
-        line_allocations_exists = line_to_refund.allocations.exists()
+        line_allocations_exists = line_to_move.allocations.exists()
         if line_allocations_exists:
             try:
-                deallocate_stock(line_to_refund, unfulfilled_to_refund)
+                deallocate_stock(line_to_move, unfulfilled_to_move)
             except AllocationError:
-                logger.warning(
-                    f"Unable to deallocate stock for line {line_to_refund.id}"
-                )
+                logger.warning(f"Unable to deallocate stock for line {line_to_move.id}")
 
-        # prepare structure which will be used to create new order event.
-        all_refunded_lines[line_to_refund.id] = (
-            unfulfilled_to_refund,
-            line_to_refund,
-        )
-
-    # update the fulfillment lines with new calues
+    # update the fulfillment lines with new values
     FulfillmentLine.objects.bulk_update(fulfillment_lines_to_update, ["quantity"])
     FulfillmentLine.objects.bulk_create(fulfillment_lines_to_create)
     OrderLine.objects.bulk_update(order_lines_to_update, ["quantity_fulfilled"])
-    return refund_amount
 
 
-def _refund_fulfillment_for_fulfillment_lines(
-    fulfillment_lines_to_refund: List[FulfillmentLine],
-    fulfillment_lines_quantities_to_refund: List[int],
-    already_refunded_lines: List[FulfillmentLine],
-    refunded_fulfillment: Fulfillment,
-    all_refunded_lines: Dict[int, Tuple[int, OrderLine]],
+def _move_fulfillment_lines_to_target_fulfillment(
```

Comment:
The main part of `_move_order_lines_to_target_fulfillment` and `_move_fulfillment_lines_to_target_fulfillment` looks very similar, maybe some part could be extracted to separate function?